kbase-extension/static/kbase/js/widgets/narrative_core/upload/stagingAreaViewer.js

```diff
@@ -600,37 +674,101 @@ define([
                 );
         },
 
+        renderImportButton: function() {
+
+            let importButton = $('<button></button>')
+            .addClass('kb-staging-table-import__button btn btn-xs btn-primary')
+            .text('Import Selected');
+
+            this.$elem.find('div.kb-staging-table-import').append(importButton);
+
+            /* 
+                By default import button is disabled until the user selects a data type 
+            */
+            this.disableImportButton();
+        },
+
+        disableImportButton: function() {
+
+            this.$elem.find('button.kb-staging-table-import__button')
+            .addClass('kb-staging-table-import__button__disabled')
+            .tooltip({
+                title: 'Select a file/s to continue.',
+                delay: {
+                    show: Config.get('tooltip').showDelay,
+                    hide: Config.get('tooltip').hideDelay
+                }
+            })
+            .off('click');
+        },
+
+        enableImportButton: function() {
+            let stagingAreaViewer = this; 
+
+            this.$elem.find('button.kb-staging-table-import__button')
+            .removeClass('kb-staging-table-import__button__disabled')
+            .tooltip('disable')
+            .unbind('click')
+            .on('click', function() {
+                stagingAreaViewer.initImport();
+            })
+            .on('keyPress', function() {
+                stagingAreaViewer.initImport();
+            });
+        },
+
+        initImport: function() {
+            let stagingAreaViewer = this; 
+
+            //get all of the selected checkbox file names and import type
+            $('input.kb-staging-table-body__checkbox-input:checked')
+            .each(function (i) {
+                const importType = $(this).attr('data-type');
+                const importFile = $(this).attr('id');
+                stagingAreaViewer.initImportApp(importType, importFile);
+            });
+          
+            stagingAreaViewer.updateView();
+        },
+
         /**
          * Initializes an import app using the given file info as input.
          * Expects 'type' to match a KBase object type string that maps onto an importer.
          * Expects 'file' to be an object with the following attributes:
          *   name = string, name of the file
          */
         initImportApp: function (type, file) {
-            var appInfo = this.uploaders.app_info[type];
+            const appInfo = this.uploaders.app_info[type];
+
             if (appInfo) {
-                var tag = APIUtil.getAppVersionTag(),
-                    fileParam = file ? file.name : '',
+                let tag = APIUtil.getAppVersionTag(),
+                    fileParam = file || '',
```

Comment:
Not sure why we would ever send an empty string here - but left it in for consistency

Reply:
Also changed from file.name -> file due to how it was being passed in

Reply:
Can you also update the docs for the function - I was wondering why `file` was not an object!